cmd/minishift/cmd/start.go

```diff
@@ -416,6 +420,20 @@ func setSubscriptionManagerParameters() {
 	cluster.RegistrationParameters.Password = viper.GetString(startFlags.Password.Name)
 }
 
+func checkRegistrationCredential(host *host.Host) {
+	needRegistration, _ := registration.NeedRegistration(host)
+	if needRegistration {
+		if cluster.RegistrationParameters.Username == "" {
+			username := pkgUtil.ReadInputFromStdin("Username to register to subscription-manager")
```

Comment:
Almost ;-) This is option 2 I mentioned, where one extends the Registrator interface. In this case imo, the method to add are:

    GetInteractiveUsernameMessage() string
    GetInteractivePasswordMessage() string

We just need the message to display to make it specific (provided UseInteractiveRegistration is _true_). The actual retrieval of the data can still happen in the command.